obspy/io/stationtxt/core.py

```diff
@@ -0,0 +1,256 @@
+
+
+def float_or_none(value):
+    if not value:
+        return None
+    else:
+        return float(value)
+
+
+def utcdatetime_or_none(value):
+    if not value:
+        return None
+    else:
+        return obspy.UTCDateTime(value)
+
+
+# The header fields of the text files at the different request levels.
+network_components = ("network", "description", "starttime", "endtime",
+                      "totalstations")
+network_types = (str, str, obspy.UTCDateTime, utcdatetime_or_none, int)
+station_components = ("network", "station", "latitude", "longitude",
+                      "elevation", "sitename", "starttime", "endtime")
+station_types = (str, str, float, float, float, str, obspy.UTCDateTime,
+                 utcdatetime_or_none)
+channel_components = ("network", "station", "location", "channel", "latitude",
+                      "longitude", "elevation", "depth", "azimuth", "dip",
+                      "sensordescription", "scale", "scalefreq", "scaleunits",
+                      "samplerate", "starttime", "endtime")
+channel_types = (str, str, str, str, float, float, float, float_or_none,
+                 float_or_none, float_or_none, str, float_or_none,
+                 float_or_none, str, float_or_none, obspy.UTCDateTime,
+                 utcdatetime_or_none)
+all_components = (network_components, station_components, channel_components)
+
+
+def unicode_csv_reader(unicode_csv_data, **kwargs):
+    csv_reader = csv.reader(utf_8_encoder(unicode_csv_data), **kwargs)
+    for row in csv_reader:
+        try:
+            yield [str(cell, "utf8") for cell in row]
+        except:
+            yield [str(cell) for cell in row]
+
+
+def utf_8_encoder(unicode_csv_data):
+    for line in unicode_csv_data:
+        if isinstance(line, native_str):
+            yield line
+        else:
+            yield line.encode('utf-8')
+
+
+def is_FDSN_station_text_file(path_or_file_object):
+    """
+    Simple function checking if the passed object contains a valid FDSN
+    station text file.
+
+    :param path_or_file_object: File name or file like object.
+    """
+    try:
+        if hasattr(path_or_file_object, "readline"):
+            cur_pos = path_or_file_object.tell()
+            first_line = path_or_file_object.readline()
+        else:
+            with open(path_or_file_object, "rt") as fh:
+                first_line = fh.readline()
+    except:
+        return False
+
+    # Attempt to move the file pointer to the old position.
+    try:
+        path_or_file_object.seek(cur_pos, 0)
+    except:
+        pass
+
+    first_line = first_line.strip()
+
+    # Attempt to decode.
+    try:
+        first_line = first_line.decode("utf-8")
+    except:
+        pass
+
+    if not first_line.startswith("#"):
+        return False
+    first_line = first_line.lstrip("#").strip()
+    if not first_line:
+        return False
+
+    # IRIS currently has a wrong header name. Just map it.
```

Comment:
Yea I also don't know how that happened. Maybe it changed between being an extension - now its an official output format of the web service.

IRIS fixed it after my report - I'm just leaving this in as I fear other services might model their own after the IRIS service and the line does not hurt.